pkg/db/interface.go

```diff
@@ -3,23 +3,38 @@ package db
 type FileID = string
 
 type CategoryID = int
+
+type WordCoordinate struct {
+	Paragraph int `json:"paragraph"`
+	Sentence  int `json:"sentence"`
+	Word      int `json:"word"`
+}
 type DocumentText struct {
-	DocumentID FileID `json:"documentID"`
-	Text       string `json:"text"`
+	DocumentID FileID         `json:"documentID"`
+	Text       string         `json:"text"`
+	FirstWord  WordCoordinate `json:"anchor"`
+	LastWord   WordCoordinate `json:"last"`
 }
 type Category struct {
-	ID    CategoryID     `json:"id"`
-	Name  string         `json:"name"`
-	Texts []DocumentText `json:"texts"`
+	ID            CategoryID     `json:"id"`
+	Name          string         `json:"name"`
+	Texts         []DocumentText `json:"texts"`
+	IsSub         bool           `json:"isSub"`
+	Subcategories []*Category    `json:"subcategories"`
```

Comment:
sneaky recursive definition ha

Reply:
Are you sure this makes sense though, since subcategories probably can't have more subcategories?

Reply:
They don't right now but past experience shows that stuff will get more complicated.

Mostly kidding. It does seem like a good time to subclass Category > Subcategory. Which go can't do? I'll rewrite using whatever the equivalent is.